src/components/NoteItem/NoteItem.tsx

```diff
@@ -1,36 +1,60 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './NoteItem.scss';
 import RemoveImg from '../../assets/remove.svg';
 import { Note } from '../../models/models';
+import RemoveModal from '../RemoveModal/RemoveModal';
 
 const NoteItem = ({
     note,
     deleteNoteItem,
     isActive,
+    darkMode,
 }: {
     note: Note;
     deleteNoteItem: (id: string) => void;
     isActive: boolean;
+    darkMode: boolean;
 }): React.ReactElement => {
+    const [modalIsOpen, setIsOpen] = useState(false);
+
+    const openModal = (): void => {
+        setIsOpen(true);
+    };
+
+    const closeModal = (): void => {
+        setIsOpen(false);
+    };
+
     const handleRemoveNoteItem = (id: string): void => {
+        closeModal();
         deleteNoteItem(id);
     };
 
+    const removeModalTitle = 'Do you really want to remove a Note?';
+
+    let noteClass = 'container-note';
```

Comment:
it looks like spaghetti code, consider using classname, or at least move it to separate function